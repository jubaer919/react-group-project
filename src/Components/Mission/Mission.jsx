import React from 'react';
import { Button, Badge } from 'react-bootstrap';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../../Redux/missions/missions';

const Mission = ({ mission }) => {
  const {
    id,
    name,
    description,
    reserved,
  } = mission;

  const dispatch = useDispatch();
  const handleClick = () => {
    if (reserved) {
      dispatch(leaveMission(id));
    } else {
      dispatch(joinMission(id));
    }
  };

  return (
    <tr key={id}>
      <th className="border fs-5 flex p-2">
        <p>{name}</p>
      </th>
      <td className="border p-2 pb-5">{description}</td>
      <td className="border p-1 align-middle">
        <Badge
          bg={reserved ? 'info' : 'secondary'}
          style={{ minWidth: '120px' }}
          className="p-3"
        >
          {reserved ? 'Active Member' : 'Not A Member'}
        </Badge>
      </td>
      <td className="p-2 border align-middle">
        <Button
          type="button"
          onClick={handleClick}
          variant={!reserved ? 'success' : 'outline-danger'}
          className="p-2 rounded text-center"
          style={{ minWidth: '120px' }}
        >
          {!reserved ? 'Join Mission' : 'Leave Mission'}
        </Button>
      </td>
    </tr>
  );
};

Mission.propTypes = {
  mission: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    description: PropTypes.string,
    reserved: PropTypes.bool,
  }).isRequired,
};

export default Mission;
