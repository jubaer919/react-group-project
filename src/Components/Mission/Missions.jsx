import React from 'react';
import Table from 'react-bootstrap/Table';

export default function Missions() {
  return (
    <div className="px-3 container">
      <Table responsive="md" className="my-5" striped>
        <thead className="p-5">
          <tr className="border">
            <th className="border">Mission</th>
            <th className="border">Description</th>
            <th className="border">Status</th>
            <th className="border"> </th>
          </tr>
        </thead>
      </Table>
    </div>
  );
}
