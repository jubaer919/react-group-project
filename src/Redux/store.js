import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './Rocket/rocketSlice';
import missionReducer from './missions/missions';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    missions: missionReducer,
  },
});

export default store;
