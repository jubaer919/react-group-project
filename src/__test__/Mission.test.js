import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Mission from '../Components/Mission/Mission';
import store from '../Redux/store';

describe('Test for Mission', () => {
  test('should render', () => {
    const msn = {
      id: 1,
      name: 'dfg',
      description: 'ghdj',
      reserved: false,
    };
    const tree = renderer.create(
      <Provider store={store}>
        <Mission mission={msn} key={1} />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
