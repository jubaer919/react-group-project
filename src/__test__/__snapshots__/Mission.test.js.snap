// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Mission should render 1`] = `
<tr>
  <th
    className="border fs-5 flex p-2"
  >
    <p>
      dfg
    </p>
  </th>
  <td
    className="border p-2 pb-5"
  >
    ghdj
  </td>
  <td
    className="border p-1 align-middle"
  >
    <span
      className="p-3 badge bg-secondary"
      style={
        Object {
          "minWidth": "120px",
        }
      }
    >
      Not A Member
    </span>
  </td>
  <td
    className="p-2 border align-middle"
  >
    <button
      className="p-2 rounded text-center btn btn-success"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "minWidth": "120px",
        }
      }
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;
